src/main.rs

```diff
@@ -259,16 +265,20 @@ impl App {
         let next_handle = self
             .handle_from_args(args)
             .or_else(|_| {
-                if args.get_flag("reverse") {
+                let reverse  = args.get_flag("reverse");
+                let fastest  = args.get_flag("fastest");
+                let shortest = args.get_flag("shortest");
+                if reverse || fastest || shortest {
                     let max_attemps = 100;
                     for _i in 0..max_attemps {
                         let handle = self.random_handle()?;
                         let clash = self.read_clash(&handle)?;
-                        if clash.is_reverse() {
-                            return Ok(handle);
-                        }
+                        if reverse  && !clash.is_reverse()  { continue; }
+                        if fastest  && !clash.is_fastest()  { continue; }
+                        if shortest && !clash.is_shortest() { continue; }
+                        return Ok(handle);
                     }
-                    Err(anyhow!(format!("No reverse clash found after {} attempts.", max_attemps)))
+                    Err(anyhow!(format!("No required clash found after {} attempts.", max_attemps)))
```

Comment:
The wording "required clash" is a bit confusing. Maybe something like "Failed to find a clash with the required modes" instead?